src/savegame.cpp

```diff
@@ -280,7 +280,9 @@ void game::load_weather(std::ifstream & fin) {
         int seed(0);
         std::stringstream liness(line);
         liness >> label >> seed;
-        weather_gen->set_seed( seed );
+        //limit seed size from old saves
+        //weather noise generator can't deal with large values
+        weather_gen->set_seed( seed % 32768 );
```

Comment:
Looks like code (and comment) duplication. The limiting should be done in the weather generator (along with the comment).

Maybe it's even enough to do it inside `weather_generator::get_weather`, the only place where the seed is actually used. That would keep the existing saves unchanged (stored seed == loaded seed) and it would allow a later change to the calculations that may work again with bigger numbers.